langchain4j-pgvector/src/main/java/dev/langchain4j/store/embedding/pgvector/JSONMetadataHandler.java

```diff
@@ -0,0 +1,87 @@
+package dev.langchain4j.store.embedding.pgvector;
+
+import dev.langchain4j.data.document.Metadata;
+import dev.langchain4j.internal.Json;
+import dev.langchain4j.store.embedding.filter.Filter;
+
+import java.sql.*;
+import java.util.*;
+
+import static dev.langchain4j.internal.ValidationUtils.ensureNotNull;
+
+/**
+ * This class handle JSON and JSONB Filter mapping
+ */
+public class JSONMetadataHandler implements MetadataHandler {
+
+    final String columnDefinition;
+    final String columnName;
+    final JSONFilterMapper filterMapper;
+    final List<String> indexes;
+
+    /**
+     * MetadataHandler constructor
+     * @param config {@link MetadataConfig} configuration
+     */
+    public JSONMetadataHandler(MetadataConfig config) {
+        this.columnDefinition = ensureNotNull(config.definition(), "Metadata definition").get(0);
+        if (config.definition().size()>1 || this.columnDefinition().contains(",")) {
+            throw new RuntimeException("Multiple columns definition are not allowed in JSON, JSONB Type");
+        }
+        this.columnName = this.columnDefinition.split(" ")[0];
+        this.filterMapper = new JSONFilterMapper(columnName);
+        this.indexes = config.indexes().orElse(Collections.emptyList());
+    }
+
+    @Override
+    public String columnDefinition() {
+        return columnDefinition;
+    }
+
+    @Override
+    public String columnsNames() {
+        return this.columnName;
+    }
+
+    @Override
+    public void createMetadataIndexes(Statement statement, String table) {
+        if (this.indexes != null && !this.indexes.isEmpty()) {
```

Comment:
```suggestion
        if (this.indexes != null || !this.indexes.isEmpty()) {
```
Was is supposed to be like this?